src/pages/token-metadata/guides/anchor/token-claimer-smart-contract.md

```diff
@@ -0,0 +1,972 @@
+    b"merkle_tree".as_ref(),
+    mint_key.as_ref(),
+    &[airdrop_state.bump],
+  ];
+
+  transfer(
+    CpiContext::new_with_signer(
+      ctx.accounts.token_program.to_account_info(),
+      Transfer {
+        from: ctx.accounts.vault.to_account_info(),
+        to: ctx.accounts.signer_ata.to_account_info(),
+        authority: airdrop_state.to_account_info(),
+      },
+      &[signer_seeds],
+    ),
+    amount,
+  )?;
+
+  Ok(())
+}
+```
+
+To ensure accuracy, the root is reconstructed from the input provided in the accounts struct and compared against the stored root on-chain to verify they match.
+
+**Step 3: Claiming**
+
+For simplicity in this example, we won’t implement concurrency. Instead, we present two possible approaches to handle claims:
+
+- Set the isClaimed flag to true and recompute the Merkle Root on-chain. The main drawback of this approach is that the account will be locked during the update, as explained in the[ Concurrent Merkle Tree section](#concurrent-merkle-tree). This limits claims to one user per block and can be implemented in the same instruction like this after the transfer:
+
+{% totem %}
+
+{% totem-accordion title="Code Example" %}
+
+```rust
+// Step 4: Update the `is_claimed` flag in the leaf
+let mut updated_leaf = Vec::new();
+updated_leaf.extend_from_slice(&ctx.accounts.signer.key().to_bytes());
+updated_leaf.extend_from_slice(&amount.to_le_bytes());
+updated_leaf.push(1u8); // isClaimed = true
+
+let updated_root: [u8; 32] = merkle_proof
+  .merklize(&updated_leaf)
+  .map_err(|_| AirdropError::InvalidProof)?
+  .try_into()
+  .map_err(|_| AirdropError::InvalidProof)?;
+
+// Step 5: Update the Merkle root in the airdrop state
+airdrop_state.merkle_root = updated_root;
+
+// Step 6: Update the airdrop state
+airdrop_state.amount_claimed = airdrop_state
+  .amount_claimed
+  .checked_add(amount)
+  .ok_or(AirdropError::OverFlow)?;
+```
+
+{% /totem-accordion %}
+
+{% /totem %}
+
+- Create a Program Derived Address (PDA) for each user after they claim their tokens. This method avoids locking the `AirdropState` account, but it requires users to pay rent for the new PDA. This can be implemented in the same instruction, we will just need to change the Account struct like this:
+
+{% totem %}
+
+{% totem-accordion title="Code Example" %}
+
+```rust
+#[derive(Accounts)]
+pub struct Claim<'info> {
+  //...
+  #[account(
+    init,
+    payer = signer,
+    seeds = [b"user_receipt".as_ref(), signer.key().to_bytes().as_ref()],
+    bump,
+    space = 8 + 32
+  )]
+  pub user_receipt: Account<'info, UserReceipt>,
+  //...
+}
+
+#[account]
+pub struct UserReceipt {
+  pub user: Pubkey,
+}
+```
+
+{% /totem-accordion %}
+
+{% /totem %}
+
+**Note**: The `user_receipt` account could be left empty to minimize rent costs. However, this makes it harder to check on the frontend if a user has already claimed their tokens. This tradeoff should be considered based on the specific requirements of the program.
```

Comment:
why does it make it harder? Based on the seeds one could check if the account or not, so no data in the account would be required?